python/pyspark/pandas/tests/data_type_ops/test_complex_ops.py

```diff
@@ -252,70 +262,76 @@ def test_invert(self):
         self.assertRaises(TypeError, lambda: ~self.psser)
 
     def test_eq(self):
-        with option_context("compute.ops_on_diff_frames", True):
-            self.assert_eq(
-                self.pser == self.other_pser, (self.psser == self.other_psser).sort_index()
-            )
-            self.assert_eq(self.pser == self.pser, (self.psser == self.psser).sort_index())
-            self.assert_eq(
-                self.struct_pser == self.struct_pser,
-                (self.struct_psser == self.struct_psser).sort_index(),
-            )
+        pdf, psdf = self.complex_pdf, self.complex_pdf
```

Comment:
Not related this change, but looks like we could also add `this_struct == this_struct` case just like array. Definately, we can add it in separate followup.

Reply:
Test cases on that are added. Thanks!